### ISSUE TITLE
The data and details methods appear to have some file permissions differences

### ISSUE THREAD

**Issue by tomharvey** [minimagick/minimagick#476]
I'm using MiniMagick on an AWS Lambda function; which has a writable `/tmp` directory, but no other directories are writable.

When I use `Image#details` I get the deprecation warning, but when I use `Image#data` I get the below. Details works fine, and the image being analysed is at `/tmp/42388e790c567ddbe3c5711f734294f9.jpg` (not the file `/tmp/mini_magick20190414-8-1lkejzs.jpg` referenced in the stacktrace.

Is there a reason why using data requires different file permissions? Or accesses some other part of the directory structure?

```
Error raised from handler method
{
  "errorMessage": "`convert /tmp/mini_magick20190414-8-1lkejzs.jpg[0] json:` failed with error:\nconvert: unable to open image `json:': Read-only file system @ error/blob.c/OpenBlob/2643.\n",
  "errorType": "Function<MiniMagick::Error>",
  "stackTrace": [
    "/var/task/vendor/bundle/ruby/2.5.0/gems/mini_magick-4.9.3/lib/mini_magick/shell.rb:17:in `run'",
    "/var/task/vendor/bundle/ruby/2.5.0/gems/mini_magick-4.9.3/lib/mini_magick/tool.rb:90:in `call'",
    "/var/task/vendor/bundle/ruby/2.5.0/gems/mini_magick-4.9.3/lib/mini_magick/tool.rb:38:in `new'",
    "/var/task/vendor/bundle/ruby/2.5.0/gems/mini_magick-4.9.3/lib/mini_magick/image/info.rb:159:in `data'",
    "/var/task/vendor/bundle/ruby/2.5.0/gems/mini_magick-4.9.3/lib/mini_magick/image/info.rb:33:in `[]'",
    "/var/task/vendor/bundle/ruby/2.5.0/gems/mini_magick-4.9.3/lib/mini_magick/image.rb:132:in `block in attribute'",
    "/var/task/lib/imgprocess/photo_metadata.rb:19:in `keywords'",
    "/var/task/lib/imgprocess/photo_metadata.rb:34:in `keywords_as_json'",
    "/var/task/lib/imgprocess/uploaded_photo.rb:94:in `to_json'",
    "/var/task/handler.rb:21:in `process'"
  ]
}```

**Comment by janko** [comment]
My only guess is that internally ImageMagick uses disk to write JSON data to, and that location isn't in the `tmp/` directory. I would consider this a bug in ImageMagick, there should be a way to tell it to store the resulting JSON in memory and not touch the disk.

**Comment by tomharvey** [comment]
Thanks @janko I am happy to look into that guess and confirm. Give me a few days.

**Comment by tomharvey** [comment]
So ... it **was** trying to write to the local file system, but not because of any error in its ability to stream the JSON output without writing. Instead it wasn't handling the `convert json:` command in version 6.7.8-9 of imagemagick and was converting to another jpg file, writing that to the filesystem.

I upgraded IM to v7 and it worked. Also seems to be working OK in version 6.9.7-4 - there was a long road to duplicate the error experienced on Lambda on a local machine!

I'll work on an AWS layer to ensure the latest version of IM is available in Lambda, or maybe the default will have moved before the deprecated `details` is fully removed.

Thanks. Closing.
